clients/csharp-dotnet-core/voice-assistant-test/tool/MainService.cs

```diff
@@ -206,80 +201,195 @@ private static async Task<bool> ProcessTestFiles(AppSettings appSettings)
                 // Keep track of high-level (summary) results for all dialogs in a single input test file. This list will be serialized to JSON as part of the overall single test report.
                 List<DialogReport> dialogReports = new List<DialogReport>();
 
-                foreach (Dialog dialog in fileContents)
+                bool sendFirst = true;
+
+                if (tests.WavAndUtterancePairs)
                 {
-                    Trace.IndentLevel = 1;
-                    if (dialog.Skip)
+                    for (int i = 0; i < 2; i++)
                     {

```

Comment:
Why do you need the loop? you alredy specify both execution blocks below, one for wav and one for text.
Also (may be related) -- you have i++ in the for statement, and also i++ at the end of the first block below. Something is fishy.
You have 3 blocks that are very similar, just differ by the folder and JSON file names. Is there away to put this block of commands in a function and call it three times with differnt folder and JSON file names?

Reply:
Done
Removed the for loop and merged the repeating code into one function - this function is called three times